First tutorial for Flows component (#6724)
SAPDocuments/Tutorials: tutorials/cp-sdk-android-flows-passcode/cp-sdk-android-flows-passcode.md

---
author_name: Haoyue Feng
author_profile: https://github.com/Fenghaoyue
title: Handle Passcode with the Flows Component
description: Learn how to use the Flows component of SAP BTP SDK for Android to handle passcode change, forgetting passcode and passcode verification scenarios.
auto_validation: true
time: 60
tags: [ tutorial>beginner, operating-system>android, topic>mobile, products>sap-business-technology-platform]
primary_tag: products>sap-btp-sdk-for-android
---

## Prerequisites

> - You completed [Try Out the SAP BTP SDK Wizard for Android](cp-sdk-android-wizard-app).

Reviewer: @FengHaoyue Can we please add the minimum SDK version the student needs to have? If the version doesn't matter then we don't need to include anything.

Author: All the tutorials are always based on the latest version of the SDK. Once a new version is released, we update the tutorial.

Reviewer: Thank you for the confirmation Haoyue.

- You completed [Get Familiar with Flows Component by Wizard Generated Application](cp-sdk-android-flows-wizard).

## Details
### You will learn
  - How to change passcode with the Flows component
  - How to handle forgetting passcode scenario with the Flows Component
  - How to verify passcode with the Flows component

---

[ACCORDION-BEGIN [Step 1: ](Handle passcode change with the Flows component)]

[OPTION BEGIN [Java]]

> 1.  Open the project you previously created using the SAP BTP SDK Wizard for Android.

Reviewer: ```suggestion
1.  Open the project you [previously created](cp-sdk-android-wizard-app) using the SAP BTP SDK Wizard for Android.
```

Reviewer: Adding a hyperlink to the required tutorial makes it easier for people to verify if they have created the project

Author: Updated for both Java and Kotlin step, thanks.

2.  In Android Studio, on Windows, press **`Ctrl+N`**, or, on a Mac, press **`command+O`**, and type **`SettingsFragment`** to open `SettingsFragment.java`.

> 3.  On Windows, press **`Ctrl+F12`**, or, on a Mac, press **`command+F12`**, and type **`onCreatePreferences`** to move to the `onCreatePreferences` method. To start the flow to change passcode, set the flow type as **`FlowType.CHANGEPASSCODE`** for the **`FlowContext`** instance, and then start the flow with this **`FlowContext`** instance. The entire process to change passcode will be handled automatically.

Reviewer: This was already set to FlowType.CHANGEPASSCODE for me.

Author: This step tells user how to start a 'change passcode' and takes the generate code as an example. The mentioned code is sample code for the flow, no need to make any change.

Reviewer: Thank you for the clarification Hao-yue. Should we then update the step as following?:

```suggestion
3.  On Windows, press **`Ctrl+F12`**, or, on a Mac, press **`command+F12`**, and type **`onCreatePreferences`** to move to the `onCreatePreferences` method. To start the flow to change passcode, the flow type must be **`FlowType.CHANGEPASSCODE`** for the **`FlowContext`** instance. Upon starting the flow with this **`FlowContext`** instance, the entire process to change passcode will be handled automatically.
```

Author: Updated for both Java and Kotlin description.
